ts/features/mvl/screens/details/components/attachment/MvlAttachmentDownload.tsx

```diff
@@ -0,0 +1,184 @@
+import { useIOBottomSheetModal } from "../../../../../../utils/hooks/bottomSheet";
+import ROUTES from "../../../../../../navigation/routes";
+import { mvlPreferencesSelector } from "../../../../store/reducers/preferences";
+
+const BOTTOM_SHEET_HEIGHT = 375;
+
+type BottomSheetProps = Readonly<{
+  /**
+   * Called on right-button press with the user's selected preferences.
+   */
+  onConfirm: (dontAskAgain: boolean) => void;
+
+  /**
+   * The user canceled the action via the UI.
+   */
+  onCancel: () => void;
+}>;
+
+const useDownloadAttachmentConfirmationBottomSheet = ({
+  onConfirm,
+  onCancel
+}: BottomSheetProps) => {
+  const [dontAskAgain, setDontAskAgain] = useState<boolean>(false);
+
+  return useIOBottomSheetModal(
+    <View>
+      <View spacer={true} />
+      <Body>
+        {i18n.t("features.mvl.details.attachments.bottomSheet.warning.body")}
+      </Body>
+      <View spacer={true} />
+      <View style={IOStyles.row}>
+        <RawCheckBox
+          checked={dontAskAgain}
+          onPress={() => setDontAskAgain(!dontAskAgain)}
+        />
+        <Body
+          style={{ paddingLeft: 8 }}
+          onPress={() => setDontAskAgain(!dontAskAgain)}
+        >
+          {i18n.t("features.mvl.details.attachments.bottomSheet.checkBox")}
+        </Body>
+      </View>
+    </View>,
+    i18n.t("features.mvl.details.attachments.bottomSheet.warning.title"),
+    BOTTOM_SHEET_HEIGHT,
+    <FooterWithButtons
+      type={"TwoButtonsInlineHalf"}
+      leftButton={{
+        ...cancelButtonProps(onCancel),
+        onPressWithGestureHandler: true
+      }}
+      rightButton={{
+        ...confirmButtonProps(() => {
+          onConfirm(dontAskAgain);
+        }, i18n.t("global.buttons.continue")),
+        onPressWithGestureHandler: true
+      }}
+    />
+  );
+};
+
+export const useMvlAttachmentDownload = (attachment: MvlAttachment) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatch = useIODispatch();
+  const { navigate } = useNavigation();
+
+  const { showAlertForAttachments } = useIOSelector(mvlPreferencesSelector);
+
+  const downloadPot = useIOSelector(state =>
+    mvlAttachmentDownloadFromIdSelector(state, attachment.id)
+  );
+
+  const openAttachment = () => {
+    const path = pot.toUndefined(downloadPot);
+
+    if (pot.isError(downloadPot)) {
+      showToast(
+        i18n.t("features.mvl.details.attachments.bottomSheet.failing.details")
+      );
+    } else if (path) {
+      if (attachment.contentType === ContentTypeValues.applicationPdf) {
+        navigate(ROUTES.MESSAGES_NAVIGATOR, {
+          screen: MVL_ROUTES.MAIN,
+          params: {
+            screen: MVL_ROUTES.ATTACHMENT,
+            params: {
+              attachment,
+              path
+            }
+          }
+        });
+      } else {
+        if (isIos) {
+          ReactNativeBlobUtil.ios.presentOptionsMenu(path);
+        } else {
+          void ReactNativeBlobUtil.android.addCompleteDownload({
```

Comment:
On Android, if I try to download a non-pdf file, I receive an error: 
```
[Error: File doesn't exist: /data/data/it.pagopa.io.app/cache/mvl/attachments/d7b23ca57da3e21cee122c377c2dc58e3fdf70397caf271b24681b1327f313c8/image_1.jpg]
```

I don't know the root cause of the error but maybe could be worth, in addition to fix this behaviour, handle the error, displaying an error toast to the user, what you think?

Reply:
Ok, could you give me some hint on your testing environment? (device and/or android version)

Reply:
I've used a physical device (`Samsung SM-M215F`) with Android 11. If can be useful, we can debug it in pair programming.

Reply:
@fabriziofff great, ping me tomorrow in the afternoon whenever you're more comfortable